### PR TITLE
adding dlq payment job production config

### PR DESCRIPTION
**Pull request by larslunde-hmcts**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by timja — CHANGES_REQUESTED** [review]
all new apps must be kustomized.

File: k8s/namespaces/fees-pay/ccpay-dlq-payment-job/prod-00.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: dead-letter-queue-process
+spec:
+  values:
+    job:
+      schedule: "0 0 * * *"
+      environment:
+        REPORT_NAME: dead-letter-queue-process
```
Comment (timja) [inline k8s/namespaces/fees-pay/ccpay-dlq-payment-job/prod-00.yaml:10]: this should be set in the base

File: k8s/prod/cluster-00-overlay/ccpay/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (timja) [inline k8s/prod/cluster-00-overlay/ccpay/kustomization.yaml:1]: this should be in a fees-pay folder to be consistent with the rest

File: k8s/namespaces/fees-pay/ccpay-dlq-payment-job/prod-00.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+        DUMMY_RESTART_VAR: true
+    global:
+      environment: prod
```
Comment (timja) [inline k8s/namespaces/fees-pay/ccpay-dlq-payment-job/prod-00.yaml:13]: this will be set via kustomize
```suggestion
```

**Review by timja — CHANGES_REQUESTED** [review]
The base needs adding in https://github.com/hmcts/cnp-flux-config/blob/dlq-payment-job-prod/k8s/prod/cluster-00-overlay/kustomization.yaml

File: k8s/namespaces/fees-pay/ccpay-dlq-payment-job/prod-00.yaml (hunk 1 above)
Comment (larslunde-hmcts) [reply k8s/namespaces/fees-pay/ccpay-dlq-payment-job/prod-00.yaml:10]: moved into base, removed env and report name as suggested.

File: k8s/prod/cluster-00-overlay/ccpay/kustomization.yaml (hunk 2 above)
Comment (larslunde-hmcts) [reply k8s/prod/cluster-00-overlay/ccpay/kustomization.yaml:1]: folder has been renamed to fees-pay.

File: k8s/namespaces/fees-pay/ccpay-dlq-payment-job/prod-00.yaml (hunk 3 above)
Comment (larslunde-hmcts) [reply k8s/namespaces/fees-pay/ccpay-dlq-payment-job/prod-00.yaml:13]: removed as suggested.

File: k8s/namespaces/fees-pay/ccpay-dlq-payment-job/dlq-payment-job.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  chart:
+    repository: https://hmctspublic.azurecr.io/helm/v1/repo/
+    name: payment-jobs
+    version: 0.0.6
+  values:
+    job:
+      image: "hmctspublic.azurecr.io/payment/jobs:prod-0a0ded6-20210716134918"   #{"$imagepolicy": "flux-system:dead-letter-queue-process"}
+    global:
+      tenantId: "531ff96d-0ae9-462a-8d2d-bec7c0b42082"
+      enableKeyVaults: true
```
[the remark this answers is not in the document]
Comment (larslunde-hmcts) [reply k8s/namespaces/fees-pay/ccpay-dlq-payment-job/dlq-payment-job.yaml:19]: removed as suggested.

File: k8s/namespaces/fees-pay/ccpay-dlq-payment-job/prod-00.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+apiVersion: helm.fluxcd.io/v1
```
Comment (timja) [inline k8s/namespaces/fees-pay/ccpay-dlq-payment-job/prod-00.yaml:1]: you need to add this file to k8s/prod/cluster-00-overlay/kustomization.yaml

in the `patchesStrategicMerge`

**Review by timja — CHANGES_REQUESTED** [review]

File: k8s/namespaces/fees-pay/ccpay-dlq-payment-job/prod-00.yaml (hunk 5 above)
Comment (larslunde-hmcts) [reply k8s/namespaces/fees-pay/ccpay-dlq-payment-job/prod-00.yaml:1]: done

**Review by timja — APPROVED** [review]

**Pull request closed without merging**